test/integration/scheduler/queue_test.go

```diff
@@ -488,16 +498,17 @@ func TestRequeueByBindFailure(t *testing.T) {
 		t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
 	}
 
-	// first binding try should fail.
-	err := wait.PollUntilContextTimeout(ctx, 200*time.Millisecond, wait.ForeverTestTimeout, false, testutils.PodSchedulingError(cs, ns, pod.Name))
+	// 1. first binding try should fail.
+	// 2. The pod should be enqueued to activeQ/backoffQ without any event.
+	// 3. The pod should be scheduled in the second binding try.
+	// Here, waiting until (3).
+	err := wait.PollUntilContextTimeout(ctx, 200*time.Millisecond, wait.ForeverTestTimeout, false, testutils.PodScheduled(cs, ns, pod.Name))
 	if err != nil {
-		t.Fatalf("Expect pod-1 to be rejected by the bind plugin: %v", err)
+		t.Fatalf("Expect pod-1 to be scheduled by the bind plugin: %v", err)
 	}
 
-	// The pod should be enqueued to activeQ/backoffQ without any event.
-	// The pod should be scheduled in the second binding try.
-	err = wait.PollUntilContextTimeout(ctx, 200*time.Millisecond, wait.ForeverTestTimeout, false, testutils.PodScheduled(cs, ns, pod.Name))
-	if err != nil {
+	// Make sure the first binding trial was failed, and this pod is scheduled at the second trial.
+	if fakeBind.counter != 1 {
```

Comment:
Yes, this will make the test passed, but I still can't figure out why we failed in the previous PR. Because the pod status's condition is still `false` now.

E.g.
if we apply below codes in the end, we'll pass usually. 🤔
```
	err = wait.PollUntilContextTimeout(ctx, 200*time.Millisecond, wait.ForeverTestTimeout, false, testutils.PodSchedulingError(cs, ns, pod.Name))
	if err != nil {
		t.Fatalf("Expect pod-1 to be rejected by the bind plugin: %v", err)
	}
```

Reply:
The PodScheduled condition is set to True by apiserver during Binding
https://github.com/kubernetes/kubernetes/blob/1a1a3d0a87530aea0294ce7ce91a79e7378b95a7/pkg/registry/core/pod/storage/storage.go#L244-L247

Reply:
Yes, so in this case, the assert should not pass(after binding), but actually, it will. Confused .. 
```
err = wait.PollUntilContextTimeout(ctx, 200*time.Millisecond, wait.ForeverTestTimeout, false, testutils.PodSchedulingError(cs, ns, pod.Name)) 
if err != nil { 
    t.Fatalf("Expect pod-1 to be rejected by the bind plugin: %v", err) 
}
```

Reply:
Sometimes we were lucky and would be able to catch the intermediate state where the pod is unschedulable.

Hence why it was flaky.

Reply:
But even you sleep for a while ,we can still get the unschedulable state. I tested with the code here: dd5ee093cf0646423dd638bbc5088ba51ee822cb